frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1600,45 +1603,51 @@ impl<T: Config> Pallet<T> {
 		{
 			Ok(())
 		} else {
-			Err("If snapshot exists, metadata and desired targets should be set too. Otherwise, none should be set.")
+			Err(DispatchError::Other("If snapshot exists, metadata and desired targets should be set too. Otherwise, none should be set."))
 		}
 	}
 
 	// [`SignedSubmissionsMap`] state check. Invariants:
 	// - All [`SignedSubmissionIndices`] are present in [`SignedSubmissionsMap`], and no more;
 	// - [`SignedSubmissionNextIndex`] is not present in [`SignedSubmissionsMap`];
 	// - [`SignedSubmissionIndices`] is sorted by election score.
-	fn try_state_signed_submissions_map() -> Result<(), &'static str> {
+	fn try_state_signed_submissions_map() -> DispatchResult {
 		let mut last_score: ElectionScore = Default::default();
 		let indices = <SignedSubmissionIndices<T>>::get();
 
 		for (i, indice) in indices.iter().enumerate() {
 			let submission = <SignedSubmissionsMap<T>>::get(indice.2);
 			if submission.is_none() {
-				return Err("All signed submissions indices must be part of the submissions map")
+				return Err(DispatchError::Other(
```

Comment:
:man_facepalming:

Reply:
I left these ones on purpose, though it would be more clear than having a `.into` after the error message 🤷